modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaEntityBrowserWidgetTest.php

```diff
@@ -0,0 +1,253 @@
+        'field_widget_display' => 'label',
+        'open' => TRUE,
+      ],
+    ])->save();
+
+    // Prepare the default view display for rendering.
+    $display = \Drupal::service('entity_display.repository')
+      ->getViewDisplay('node', 'page')
+      ->setComponent('featured_media_field', $view_display_options);
+    $display->save();
+
+    $this->drupalLogin($this->drupalCreateUser([], '', TRUE));
+  }
+
+  /**
+   * Tests the featured media entity browser widget.
+   */
+  public function testFeaturedMediaEntityBrowserWidget(): void {
+    $this->drupalGet('node/add/page');
+
+    // Assert that all the entity browser elements are displayed.
+    $this->assertSession()->buttonExists('Select images');
+    $this->assertSession()->fieldExists('Caption');
+    $this->assertSession()->pageTextContains('The caption that goes with the referenced media.');
+    $this->assertSession()->buttonExists('Add another item');
+
+    // Select one media image from the entity browser.
+    $this->getSession()->getPage()->pressButton('Select images');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->switchToIFrame('entity_browser_iframe_test_images');
+    $this->getSession()->getPage()->checkField('entity_browser_select[media:1]');
+    $this->getSession()->getPage()->pressButton('Select image');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Assert the image was selected and the widget shows the proper buttons.
+    $this->assertSession()->pageTextContains('Image 1');
+    $this->assertSession()->buttonNotExists('Select images');
+    $this->assertSession()->buttonExists('Remove');
+    $this->assertSession()->buttonExists('Edit');
+
+    // Add the other media image item.
+    $this->getSession()->getPage()->pressButton('Add another item');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->getPage()->pressButton('Select images');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->switchToIFrame('entity_browser_iframe_test_images');
+    $this->getSession()->getPage()->checkField('entity_browser_select[media:2]');
+    $this->getSession()->getPage()->pressButton('Select image');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Image 2');
+    $this->assertSession()->buttonNotExists('Select images');
+
+    // Check that 'Image 1' media item is placed before 'Image 2'.
+    $this->assertOrderInPage(['Image 1', 'Image 2']);
+
+    // Fill in the other fields and save the node.
+    $this->getSession()->getPage()->fillField('featured_media_field[0][caption]', 'Image 1 caption');
+    $this->getSession()->getPage()->fillField('featured_media_field[1][caption]', 'Image 2 caption');
+    $this->getSession()->getPage()->fillField('Title', 'Test entity browser widget');
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert that the values were saved.
```

Comment:
Well, if there was an error, nothing will happen and it will be a saved positive. Assert the save message is printed on the page or reload.

Load also the node and assert the values.